### PR TITLE
#162163003 Build out REST endpoints for Article  

### PR DESCRIPTION
**Pull request by katunold**
#### What does this PR do?
* Implement **CRUD** on article endpoint in ah REST API
* Write tests for the article app
#### Description of Task to be completed?
*Have the following endpoints working*
1. List Articles
`GET /api/articles/` To return a list of all articles
2. Get specific Article
`GET /api/articles/slug` To return a specific article basing on the slug provided
3. Create Article
`POST /api/articles/` To create and return the article created
Example request body
```
{
  "article": {
    "title": "Who is he?",
    "description": "He has a bald head",
    "body": "It is believed he will rule Uganda to heaven"
  }
}
```
4. Update Article
`PUT /api/articles/slug/` To make changes to any fields that are input by the user while creating the article. that is if the user posted the article
```
{
  "article": {
    "title": "Who is he? a god or demigod?"
  }
}
```
5. Delete Article
`DELETE /api/articles/slug` To delete an article only posted by the user
#### How should this be manually tested?
After cloning the repo, cd into it and follow the steps 
* In terminal, type `git checkout ft-create-article-162163003` to switch to my branch
* Open your prefered editor and go to the settings. py file in the project
* In the settings.py file, edit the value assigned to a password in Database to what you use locally on your computer
* Run the server using python manage.py runserver
* Using postman test every endpoint above using these headers:
_key_: content-type _value_: application/json
_key_: Authorization _value_: Token **your token**
#### What are the relevant pivotal tracker stories?
[#162163003](https://www.pivotaltracker.com/story/show/162163003)

### CODE REVIEW

**Review by kizzanaome — APPROVED** [review]

**Review by okellogabrielinnocent — APPROVED** [review]
Work well laid

File: authors/apps/article/models.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+class Article(models.Model):
+
+    # Model for an article
+    author = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="articles",
+        null=True
```
Comment (arthurarty) [inline authors/apps/article/models.py:20]: Null = true? Wouldnt this mean that an article can exist without an author ?

File: authors/settings.py [hunk 2]
```diff
@@ -91,7 +92,7 @@
         'ENGINE': 'django.db.backends.postgresql_psycopg2',
         'NAME': os.environ.get('DB_NAME', 'ahdb'),
         'USER': os.environ.get('DB_USER', 'postgres'),
-        'PASSWORD': os.environ.get('PASSWORD', '1460'),
+        'PASSWORD': os.environ.get('PASSWORD', ''),
```
Comment (arthurarty) [inline authors/settings.py:95]: good fix. I wonder who had once again added a password here. @kizzanaome ?

**Review by arthurarty — APPROVED** [review]
Otw it looks good to go.

**Pull request merged**